src/location-menu/location-menu.component.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { createErrorHandler } from "@openmrs/esm-error-handling";
+import {
+  getLoginLocations,
+  setSessionLocation
+} from "../location-menu/location-menu.resource";
+import style from "./location-menu.css";
+
+export default function LocationMenu(props: LocationMenuProps) {
+  const [loginLocations, setLoginLocations] = React.useState([]);
+  const [newLocation, setNewLocation]: [
+    Location,
+    (l: Location) => void
+  ] = React.useState(null);
+
+  React.useEffect(() => {
+    const sub = getLoginLocations().subscribe(
+      locations => setLoginLocations(locations),
+      createErrorHandler()
+    );
+    return () => sub.unsubscribe();
+  }, []);
+
+  React.useEffect(() => {
+    const abortController = new AbortController();
```

Comment:
So the abort controller is a somethign that's kind of use like an abort-handling callback? Is this a class provided by a library?  Where is it defined?

Reply:
Yes; https://developer.mozilla.org/en-US/docs/Web/API/AbortController